### PR TITLE
SDIT-937 Add health test and additional config

### PR DESCRIPTION
**Pull request by karenmillermoj**
This PR just grew - so will need additional tidy up

### CODE REVIEW

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/integration/health/HealthCheckTest.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      WireMock.aResponse()
+        .withHeader("Content-Type", "application/json")
+        .withBody(if (status == 200) "pong" else "some error")
+        .withStatus(status),
+    ),
+  )
+}
+
+
+ */
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/integration/health/HealthCheckTest.kt:103]: Will remove in next PR

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/config/ApplicationInsightsConfiguration.kt [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
 /**
  * TelemetryClient gets altered at runtime by the java agent and so is a no-op otherwise
  */
 @Configuration
-class ApplicationInsightsConfiguration {
-  @Bean
-  fun telemetryClient(): TelemetryClient = TelemetryClient()
-}
-
+class ApplicationInsightsConfiguration
```
Comment (petergphillips) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/config/ApplicationInsightsConfiguration.kt:10]: guess you don't actually need lines 9, 10 or 11 now (11 is in common)

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/OpenAPIConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class OpenAPIConfiguration {
+  @Autowired(required = false)
+  private val buildProperties: BuildProperties? = null
+
+  @Bean
+  fun api(): OpenAPI {
+    return OpenAPI()
+      .info(
+        Info().title("Prisoner search API Documentation")
+          .description("API for searching for offenders in Delius.")
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/OpenAPIConfiguration.kt:21]: would have been better to copy this file from indexer instead - don't like the nullable `BuildProperties` etc

Comment (karenmillermoj) [reply hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/OpenAPIConfiguration.kt:21]: Will address the comments in a separate PR

**Pull request merged**